File: mobile-rn/src/screens/AuthScreen.js
import { useState } from "react";
import { KeyboardAvoidingView, Platform, Pressable, ScrollView, StyleSheet, Text, View } from "react-native";
import { colors } from "../theme";
import { Field, PrimaryButton } from "../components/ui";
import { api, ApiError } from "../api";
import { useAuth } from "../context/AuthContext";

export default function AuthScreen({ route, navigation }) {
  const { setAuth } = useAuth();
  const [mode, setMode] = useState(route.params?.mode || "login");
  const [phone, setPhone] = useState("");
  const [pin, setPin] = useState("");
  const [storeName, setStoreName] = useState("");
  const [error, setError] = useState("");
  const [submitting, setSubmitting] = useState(false);

  async function handleSubmit() {
    setError("");
    setSubmitting(true);
    try {
      const data = mode === "login" ? await api.login(phone.trim(), pin.trim()) : await api.register(phone.trim(), pin.trim(), storeName.trim());
      setAuth({ token: data.token, phone: data.phone, maskedPhone: data.maskedPhone, storeName: data.storeName || "" });
      navigation.goBack();
    } catch (err) {
      setError(err instanceof ApiError ? err.message : "Une erreur s'est produite.");
    } finally {
      setSubmitting(false);
    }
  }

  return (
    <KeyboardAvoidingView style={styles.flex} behavior={Platform.OS === "ios" ? "padding" : undefined}>
      <ScrollView contentContainerStyle={styles.container}>
        <View style={styles.tabs}>
          <Pressable style={[styles.tab, mode === "login" && styles.tabActive]} onPress={() => setMode("login")}>
            <Text style={[styles.tabText, mode === "login" && styles.tabTextActive]}>Connexion</Text>
          </Pressable>
          <Pressable style={[styles.tab, mode === "register" && styles.tabActive]} onPress={() => setMode("register")}>
            <Text style={[styles.tabText, mode === "register" && styles.tabTextActive]}>Inscription</Text>
          </Pressable>
        </View>

        <Text style={styles.sub}>
          {mode === "login"
            ? "Connectez-vous avec votre numéro de téléphone vendeur et votre code PIN."
            : "Inscrivez-vous avec un numéro de téléphone et un code PIN de 4 à 6 chiffres — aucun e-mail requis."}
        </Text>

        <Field placeholder="Numéro de téléphone — ex. +221771234567" keyboardType="phone-pad" value={phone} onChangeText={setPhone} style={styles.gap} />
        <Field
          placeholder="Code PIN (4 à 6 chiffres)"
          keyboardType="number-pad"
          secureTextEntry
          maxLength={6}
          value={pin}
          onChangeText={setPin}
          style={styles.gap}
        />
        {mode === "register" && (
          <Field placeholder="Nom de la boutique (facultatif)" maxLength={40} value={storeName} onChangeText={setStoreName} style={styles.gap} />
        )}

        <PrimaryButton
          title={mode === "login" ? "Se connecter" : "Créer un compte"}
          onPress={handleSubmit}
          loading={submitting}
          style={styles.gap}
        />
        {!!error && <Text style={styles.error}>{error}</Text>}
      </ScrollView>
    </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  flex: { flex: 1, backgroundColor: colors.panel },
  container: { padding: 22 },
  tabs: { flexDirection: "row", gap: 6, marginBottom: 14 },
  tab: { flex: 1, paddingVertical: 8, borderRadius: 20, borderWidth: 1, borderColor: colors.hairline, alignItems: "center" },
  tabActive: { borderColor: colors.amber, backgroundColor: "rgba(232,163,61,0.08)" },
  tabText: { color: colors.textDim, fontSize: 12 },
  tabTextActive: { color: colors.amber },
  sub: { color: colors.textDim, fontSize: 12, lineHeight: 18, marginBottom: 16 },
  gap: { marginBottom: 10 },
  error: { color: colors.error, fontSize: 11.5, marginTop: 8 }
});

File: mobile-flutter/lib/screens/auth_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../services/api.dart';
import '../services/auth_provider.dart';
import '../theme/colors.dart';

class AuthScreen extends StatefulWidget {
  final bool startInRegister;
  const AuthScreen({super.key, this.startInRegister = false});

  @override
  State<AuthScreen> createState() => _AuthScreenState();
}

class _AuthScreenState extends State<AuthScreen> {
  late bool _isLogin = !widget.startInRegister;
  final _phoneCtrl = TextEditingController();
  final _pinCtrl = TextEditingController();
  final _storeNameCtrl = TextEditingController();
  String _error = '';
  bool _submitting = false;

  Future<void> _submit() async {
    setState(() {
      _error = '';
      _submitting = true;
    });
    try {
      final data = _isLogin
          ? await Api.instance.login(_phoneCtrl.text.trim(), _pinCtrl.text.trim())
          : await Api.instance.register(_phoneCtrl.text.trim(), _pinCtrl.text.trim(), _storeNameCtrl.text.trim());
      if (!mounted) return;
      await context.read<AuthProvider>().setAuth(SellerAuth.fromJson(data));
      if (!mounted) return;
      Navigator.of(context).pop();
    } catch (err) {
      setState(() => _error = err is ApiException ? err.message : 'Une erreur s\'est produite.');
    } finally {
      if (mounted) setState(() => _submitting = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Compte vendeur')),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(22),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            Row(
              children: [
                Expanded(
                  child: OutlinedButton(
                    style: OutlinedButton.styleFrom(
                      side: BorderSide(color: _isLogin ? SnapyColors.amber : SnapyColors.hairline),
                      foregroundColor: _isLogin ? SnapyColors.amber : SnapyColors.textDim,
                    ),
                    onPressed: () => setState(() => _isLogin = true),
                    child: const Text('Connexion'),
                  ),
                ),
                const SizedBox(width: 6),
                Expanded(
                  child: OutlinedButton(
                    style: OutlinedButton.styleFrom(
                      side: BorderSide(color: !_isLogin ? SnapyColors.amber : SnapyColors.hairline),
                      foregroundColor: !_isLogin ? SnapyColors.amber : SnapyColors.textDim,
                    ),
                    onPressed: () => setState(() => _isLogin = false),
                    child: const Text('Inscription'),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 16),
            Text(
              _isLogin
                  ? 'Connectez-vous avec votre numéro de téléphone vendeur et votre code PIN.'
                  : 'Inscrivez-vous avec un numéro de téléphone et un code PIN de 4 à 6 chiffres — aucun e-mail requis.',
              style: const TextStyle(color: SnapyColors.textDim, fontSize: 12, height: 1.4),
            ),
            const SizedBox(height: 16),
            TextField(
              controller: _phoneCtrl,
              keyboardType: TextInputType.phone,
              decoration: const InputDecoration(hintText: 'Numéro de téléphone — ex. +221771234567'),
            ),
            const SizedBox(height: 10),
            TextField(
              controller: _pinCtrl,
              keyboardType: TextInputType.number,
              obscureText: true,
              maxLength: 6,
              decoration: const InputDecoration(hintText: 'Code PIN (4 à 6 chiffres)'),
            ),
            if (!_isLogin) ...[
              const SizedBox(height: 10),
              TextField(
                controller: _storeNameCtrl,
                maxLength: 40,
                decoration: const InputDecoration(hintText: 'Nom de la boutique (facultatif)'),
              ),
            ],
            const SizedBox(height: 6),
            ElevatedButton(
              onPressed: _submitting ? null : _submit,
              child: _submitting
                  ? const SizedBox(width: 20, height: 20, child: CircularProgressIndicator(strokeWidth: 2, color: SnapyColors.amberOn))
                  : Text(_isLogin ? 'Se connecter' : 'Créer un compte'),
            ),
            if (_error.isNotEmpty) ...[
              const SizedBox(height: 8),
              Text(_error, style: const TextStyle(color: SnapyColors.error, fontSize: 11.5)),
            ],
          ],
        ),
      ),
    );
  }
}
